src/main/java/com/example/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -75,7 +75,7 @@ private CartItem createCartItem(CartItemRequestDto cartItemRequestDto,
         return cartItem;
     }
 
-    private User getPrincipal(Authentication authentication) {
+    public User getPrincipal(Authentication authentication) {

```

Comment:
ok, I see that you want to use it in different services but in this case, which relation does this method have to ShoppingCartService?

Reply:
should I duplicate it, because I need to do same action in OrderService and ShoppingCartService?

Reply:
It’s up to you. You can move it to UserService